mantidimaging/core/utility/execution_timer.py

```diff
@@ -35,21 +35,21 @@ def __init__(self, msg: str = 'Elapsed time', logger: Logger = perf_logger):
         self.time_start: float | None = None
         self.time_end: float | None = None
 
-    def __str__(self):
+    def __str__(self) -> str:
         prefix = f'{self.msg}: ' if self.msg else ''
         sec = self.total_seconds
         return f'{prefix}{sec if sec else "unknown"} seconds'
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         self.time_start = time.monotonic()
         self.time_end = None
 
-    def __exit__(self, *args):
+    def __exit__(self, *args) -> None:
         self.time_end = time.monotonic()
         self.logger.info(str(self))
 
     @property
-    def total_seconds(self):
+    def total_seconds(self) -> float | None:
```

Comment:
I'd be happy to change the `else None` to `else 0`, so that the return type can just be `float`